pennylane/wires.py

```diff
@@ -0,0 +1,322 @@
+        if not isinstance(wires, Wires):
+            raise WireError(
+                "expected a `pennylane.wires.Wires` object; got {} of type {}.".format(
+                    wires, type(wires)
+                )
+            )
+
+        diff = [w for w in self.wire_list if w not in wires.wire_list]
+        return Wires(diff)
+
+    def get_indices(self, wires):
+        """
+        Return the indices of the wires in this Wires object.
+
+        For example,
+
+        >>> wires1 =  Wires([4, 0, 1])
+        >>> wires2 = Wires([1, 4])
+        >>> wires1.get_indices(wires2)
+        [2, 0]
+
+        Args:
+            wires (Wires): A Wires class object
+
+        Returns:
+            List: index list
+        """
+
+        if not isinstance(wires, Wires):
+            raise WireError(
+                "expected a `pennylane.wires.Wires` object; got {} of type {}.".format(
+                    wires, type(wires)
+                )
+            )
+
+        return [self.index(w) for w in wires]
+
+    def injective_map_exists(self, wires):
+        """
+        Checks that there is an injective mapping between ``wires`` and this object.
+
+        Since :class:`pennylane.wires.Wires`` objects are by definition collections of unique elements, we
+        only need to check the length of both sequences.
+
+        Args:
+            wires (Wires): A Wires class object
+
+        Returns:
+            bool: whether injective mapping exists or not
+        """
+
+        if not isinstance(wires, Wires):
+            raise WireError(
+                "expected a `pennylane.wires.Wires` object; got {} of type {}.".format(
+                    wires, type(wires)
+                )
+            )
+
+        if len(self.wire_list) == len(wires):
+            return True
+        else:
+            return False
+
+    def select(self, indices, periodic_boundary=False):
+        """
+        Returns a subset of the Wires specified by the 'indices'.
+
+        For example:
+
+        >>> wires = Wires([4, 0, 1, 5, 6])
+        >>> wires.select([2, 3, 0]) == Wires([1, 5, 4])
+        True
+
+        >>> wires.select(1) == Wires([0])
+        True
+
+        Args:
+            indices (List[int] or int): indices or index of the wires we want to select
+
+        Returns:
+            Wires: subset of wires
+        """
+
+        if isinstance(indices, int):
+            indices = [indices]
+
+        if periodic_boundary:
+            # replace indices by their modulo
+            indices = [i % len(self) for i in indices]
+
+        for i in indices:
+            if i > len(self):
+                raise WireError(
+                    "cannot select wire at index {} from {} wires.".format(i, len(self))
+                )
+
+        subset = [self.wire_list[i] for i in indices]
+        return Wires(subset)
+
+    def select_random(self, n_samples, seed=None):
+        """
+        Returns a randomly sampled subset of Wires of length 'n_samples'.
```

Comment:
😆 What's the use case of this one?

Reply:
RandomLayers template. Or any future templates where you safely need to sample wires - which has caused bugs before.